### ISSUE TITLE
pull --rebase --autostash does NOT the same thing

### ISSUE THREAD

**Issue by ScreamingDev** [aanand/git-up#121]
What I used git up for:

- See if every branch is in a valid state.
- Valid means:
  - Nothing to stash
  - Remote / tracking branch is set correctly
  - No commit behind or in front

What rebase autostash does:

- Nagging about missing tracking branch and aborting.

So in general I do no longer get a sum up of all the problem to handle them but instead I get one problem after another which is so damn slowing down.

Would you recover working on this tool?
Shall I?
Are there other?

**Comment by aanand** [comment]
I'm sorry that your use case isn't met, but I will not be reviving git-up as long as I don't personally need it. There are probably other tools for doing the same thing, but I don't know of any off the top of my head.

**Comment by MatthewBooth** [comment]
I agree with @ScreamingDev it definitely doesn't do the same thing. However, @aanand you needn't worry, as this has been working for me for years. It's _the_ solution to keeping my repos up to date.

**Comment by 0x-2a** [comment]
Here's my latest workaround... create a script to do this then alias it in your git config...

Copy the gist somewhere into `git-up.sh`, then `chmod +x git-up.sh`
https://gist.github.com/0x-2a/295cbe513085413d4ca245ce112da2ea

Open your `~/.gitconfig` with your preferred editor, add this to the bottom (or to the alias section if you already have it)
```
[alias]
  up = !zsh /path-to-your-git-up-sh-file/git-up.sh
```

**Comment by seanadkinson** [comment]
@0x-2a That gist is gone now.  Might you still have the script?

**Comment by 0x-2a** [comment]
@seanadkinson oh yeah, looks like the link didn't survive my username change -- updated to:

https://gist.github.com/0x-2a/295cbe513085413d4ca245ce112da2ea

I still get some occasional oddities with it, but mostly has been working ok.
